unity-environment/Assets/ML-Agents/Examples/WallJump/Scripts/WallJumpAgent.cs

```diff
@@ -0,0 +1,368 @@
+    // less floaty
+    public float fallingForce;
+    // Use to check the coliding objects
+    public Collider[] hitGroundColliders = new Collider[3];
+    Vector3 jumpTargetPos;
+    Vector3 jumpStartingPos;
+
+    string[] detectableObjects;
+
+    void Awake()
+    {
+        academy = FindObjectOfType<WallJumpAcademy>();
+    }
+
+    public override void InitializeAgent()
+    {
+        configuration = Random.Range(0, 5);
+        detectableObjects = new string[] { "wall", "goal", "block" };
+
+        agentRB = GetComponent<Rigidbody>();
+        shortBlockRB = shortBlock.GetComponent<Rigidbody>();
+        spawnAreaBounds = spawnArea.GetComponent<Collider>().bounds;
+        groundRenderer = ground.GetComponent<Renderer>();
+        groundMaterial = groundRenderer.material;
+
+        spawnArea.SetActive(false);
+    }
+
+
+    // Begin the jump sequence
+    public void Jump()
+    {
+
+        jumpingTime = 0.2f;
+        jumpStartingPos = agentRB.position;
+    }
+
+    /// <summary>
+    /// Does the ground check.
+    /// </summary>
+    /// <returns><c>true</c>, if the agent is on the ground, 
+    /// <c>false</c> otherwise.</returns>
+    /// <param name="boxWidth">The width of the box used to perform 
+    /// the ground check. </param>
+    public bool DoGroundCheck(float boxWidth)
+    {
+        hitGroundColliders = new Collider[3];
+        Physics.OverlapBoxNonAlloc(
+            gameObject.transform.position + new Vector3(0, -0.05f, 0),
+                            new Vector3(boxWidth/2f, 0.5f, boxWidth / 2f),
+                            hitGroundColliders,
+                            gameObject.transform.rotation);
+        bool grounded = false;
+        foreach (Collider col in hitGroundColliders)
+        {
+
+            if (col != null && col.transform != this.transform &&
+                (col.CompareTag("walkableSurface") ||
+                col.CompareTag("block") ||
+                 col.CompareTag("wall")))
+            {
+                grounded = true; //then we're grounded
+                break;
+            }
+        }
+        return grounded;
+    }
+
+
+    /// <summary>
+    /// Moves  a rigidbody towards a position smoothly.
+    /// </summary>
+    /// <param name="targetPos">Target position.</param>
+    /// <param name="rb">The rigidbody to be moved.</param>
+    /// <param name="targetVel">The velocity to target during the
+    ///  motion.</param>
+    /// <param name="maxVel">The maximum velocity posible.</param>
+    void MoveTowards(
+        Vector3 targetPos, Rigidbody rb, float targetVel, float maxVel)
+    {
+        Vector3 moveToPos = targetPos - rb.worldCenterOfMass;  
+        Vector3 velocityTarget = moveToPos * targetVel * Time.fixedDeltaTime;
+        if (float.IsNaN(velocityTarget.x) == false)
+        {
+            rb.velocity = Vector3.MoveTowards(
+                rb.velocity, velocityTarget, maxVel);
+        }
+    }
+
+    public void RayPerception(float rayDistance,
```

Comment:
seems like this could benefit from @awjuliani PR where he refactored the perception change ... I would incorporate and then send out another PR.

Reply:
Not until merged into development-0.3